contracts/contracts/OpenSignal.sol

```diff
@@ -98,7 +98,7 @@ contract OpenSignal is ERC2771Context, ReentrancyGuard {
         nativeToken.safeTransferFrom(address(this), _msgSender(), project.selfStake);
     }
 
-    function addSignal(bytes32 id, uint256 amount) external {
+    function addSignal(bytes32 id, uint256 amount, uint256 minShares) external {
```

Comment:
so minShares is just a number that the user inputs and says that they want this minimum number of shares OR they will not purchase the shares?

Reply:
Yep